theano/gpuarray/subtensor.py

```diff
@@ -582,7 +584,14 @@ class GpuAdvancedIncSubtensor1(Op):
     """
     _f16_ok = True
     __props__ = ('inplace', 'set_instead_of_inc')
-    params_type = gpu_context_type
+    params_type = ParamsType(inplace=bool_t,
+                             set_instead_of_inc=bool_t,
+                             context=gpu_context_type,
+                             # following params are for c_init_code_struct.
+                             ndim_input_0=size_t,
+                             ndim_input_1=size_t,
+                             typecode_input_0=int_t,
+                             typecode_input_1=int_t)
```

Comment:
I think we should see the ParamsType as being the props + context and nothing more. I think all what can be computed at run time should be there at run time. This make the code simpler/more readable I think. If you think otherwise, tell it.